storage/range_command.go

```diff
@@ -592,11 +592,11 @@ func (r *Range) InternalPushTxn(batch engine.Engine, ms *proto.MVCCStats, args *
 		pusherWins = false
 	} else if reply.PusheeTxn.Priority < priority ||
 		(reply.PusheeTxn.Priority == priority && args.Txn != nil &&
-			args.Txn.Timestamp.Less(reply.PusheeTxn.Timestamp)) {
-		// Pusher wins based on priority; if priorities are equal, order
-		// by lower txn timestamp.
+			bytes.Compare(args.Txn.ID, reply.PusheeTxn.ID) > 0) {
```

Comment:
It's actually not inconceivable that we could have clocks syncing considering that we update clocks based on highest clock seen so far, and with the wall time in nanos sufficiently advanced, there's actually a pretty high likelihood that we'd see the logical times march ahead closely enough to overlap on two nodes (in case of tests, we have exact same clock, so I'm not quite sure this would happen).

In any case, I like this better, so keep this change.